libnavigation-core/src/main/java/com/mapbox/navigation/core/MapboxNavigation.kt

```diff
@@ -662,6 +663,74 @@ class MapboxNavigation(
         )
     }
 
+    /**
+     * Create and cache a user feedback about an issue or problem with the Navigation SDK.
+     *
+     * Instead of sending the feedback, this API will cache the feedback in the Navigation SDK. The cached feedbacks can be retrieved by calling [getCachedUserFeedback] so the user can update and send them in a later time. To send the cached feedbacks, please use [postCachedUserFeedback].
+     *
+     * If you want to send the feedback directly without caching it, you should use [postUserFeedback].
+     *
+     * @param feedbackType one of [FeedbackEvent.Type]
+     * @param description description message
+     * @param feedbackSource one of [FeedbackEvent.Source]
+     * @param screenshot encoded screenshot (optional)
+     * @param feedbackSubType array of [FeedbackEvent.Description] (optional)
+     * @param appMetadata [AppMetadata] information (optional)
+     *
+     * @see [postUserFeedback]
+     * @see [getCachedUserFeedback]
+     * @see [postCachedUserFeedback]
+     */
+    fun cacheUserFeedback(
```

Comment:
why do we need both `cacheUserFeedback ()` and `postUserFeedback` ?
what usecase do we have for each of them?

Reply:
in short, feedback 2.0 flow has different concept of 1tap and 2tap flow.
* for 1tap flow, user will use the `postUserFeedback` so the feedback will be queued to send.
* for 2tap flow, user will use the `cacheUserFeedback` so the feedback is cached and will be able to do update to it in a later time.